content/learn/book/ecs/_index.md

```diff
@@ -0,0 +1,114 @@
+* For finer control of the order of execution within a scheduler
+  - You can create as many NEW stages in the scheduler as you like
+  - New stages are added immediately *before* or *after* an existing stage (system defined or custom)
+* You can emit and consume custom events
+* You can add new **Components** and **Systems** to a running {{rust_type(type="struct", crate="bevy_app", name="App", no_mod=true, plural=false)}}, i.e. after `.run()`
+  - with `commands:` {{rust_type(type="struct", crate="bevy_ecs", name="Commands", no_mod=true, plural=false)}}
+* {{rust_type(type="struct", crate="bevy_app", name="App", no_mod=true, plural=true)}} also have a mechanism for exiting gracefully
+
+## Create Project
+
+First [create](@/learn/book/getting-started/setup/_index.md#create-a-new-rust-executable-project) a new project called `bevy_ecs_tut` and [add](@/learn/book/getting-started/setup/_index.md#add-bevy-to-your-project-s-cargo-toml) <img src="/assets/bevy_logo_dark.svg" style="height: 1em;margin-bottom:-0.1em" alt="BEVY"/> as a dependency.
+
+If you don't know or remember the current version of Bevy, `cargo` can fetch it for you fast.
+```bash
+$ cargo search bevy
+bevy = "0.1.3"                  # A refreshingly simple data-driven ...
+```
+{{caption(ref=3.1, desc="Truncated output of cargo search")}}
+
+## AppBuilder
+
+{{fileref(name="src/main.rs")}}
+
+```rs
+use bevy::prelude::*;
+
+fn main() {
+    App::build()
+        .init_resource::<GameState>()
+        .add_startup_system(print_initial_system.system())
+        .run();
+}
+
+#[derive(Default)]
+struct GameState {
+    current_round: usize,
+    total_players: usize,
+    winning_player: Option<String>,
+}
+
+fn print_initial_system(state: Res<GameState>) {
+    println!("Welcome to the Game!
+The default GameState is:\n\tCurrent Round: {}
+\tTotal Players: {}\n\tWinning Player: {:?}",
+          state.current_round, state.total_players, state.winning_player);
+}
+```
+{{caption(ref=3.2, desc="A simple App")}}
+
+At the risk of stating the obvious.
+```bash
+$ cargo run
+
+Welcome to the Game!
+The default GameState is:
+        Current Round: 0
+        Total Players: 0
+        Winning Player: None
+```
+{{caption(ref=3.3, desc="First game output")}}
+
+The values of `GameState` come from the derived `Default` implementation.
+
+{{rust_type(type="struct" crate="bevy_app", name="App" method="build" no_struct=false)}} makes an {{rust_type(type="struct", crate="bevy_app", name="AppBuilder", no_mod=true, plural=false)}} for us, on which we can call inline methods to add **Systems**, **Components**, and configuration.
+
+{{rust_type(type="struct" crate="bevy_app", name="AppBuilder" method="init_resource" no_struct=false)}} inserts `GameState` into {{rust_type(type="struct", crate="bevy_app", name="App", no_mod=true, plural=false)}}::resources<{{rust_type(type="struct" crate="bevy_ecs", name="Resources", no_mod=false, plural=false)}}>, making `GameState` a **Component**.
```

Comment:
The formatting here feels a bit awkward / hard to parse:
![image](https://user-images.githubusercontent.com/2694663/92047057-f99d2b80-ed38-11ea-9f26-d0f240c7204e.png)

I think we should make that a single "inline code block" and then add separate references after the fact.